### PR TITLE
The worker and the delivery is split into several classes

### PR DESCRIPTION
**Pull request by annando**
To finalize the bulk delivery that had been added as a performance improvement, in the worker cron we now directly deliver messages or perform a requeue if needed. Since the classes already were huge, they have been split.

### CODE REVIEW

**Comment by annando** [comment]
@nupplaphil the license check seems to fail with command line commands, since the code starts with `#!/usr/bin/env php` but not the license. Can you check that?

File: src/Core/Worker/Daemon.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+use Friendica\App\Mode;
+use Friendica\Core\Logger;
+use Friendica\DI;
+
+/**
+ * Contains the class for the worker background job processing
+ */
+class Daemon
+{
+	private static $daemon_mode = null;
```
Comment (MrPetovan) [inline src/Core/Worker/Daemon.php]:
```suggestion
	private static $mode = null;
```

File: src/Core/Worker/Daemon.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		if (!is_readable($pidfile)) {
+			// No pid file. We assume that the daemon had been intentionally stopped.
+			self::$daemon_mode = false;
+			return false;
+		}
+
+		$pid     = intval(file_get_contents($pidfile));
+		$running = posix_kill($pid, 0);
+
+		self::$daemon_mode = $running;
```
Comment (MrPetovan) [inline src/Core/Worker/Daemon.php]:
```suggestion
		if (!is_null(self::$mode)) {
			return self::$mode;
		}

		if (DI::mode()->getExecutor() == Mode::DAEMON) {
			return true;
		}

		$mode = DI::config()->get('system', 'worker_daemon_mode', false, true);
		if ($mode) {
			return $mode;
		}

		if (!function_exists('pcntl_fork')) {
			self::$mode = false;
			return false;
		}

		$pidfile = DI::config()->get('system', 'pidfile');
		if (empty($pidfile)) {
			// No pid file, no daemon
			self::$mode = false;
			return false;
		}

		if (!is_readable($pidfile)) {
			// No pid file. We assume that the daemon had been intentionally stopped.
			self::$mode = false;
			return false;
		}

		$pid     = intval(file_get_contents($pidfile));
		$running = posix_kill($pid, 0);

		self::$mode = $running;
```

File: src/Core/Worker/Daemon.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+		}
+
+		if (DI::mode()->getExecutor() == Mode::DAEMON) {
+			return true;
+		}
+
+		$daemon_mode = DI::config()->get('system', 'worker_daemon_mode', false, true);
+		if ($daemon_mode) {
+			return $daemon_mode;
+		}
```
Comment (MrPetovan) [inline src/Core/Worker/Daemon.php:53]: Is it expected not to have any caching here?

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Core/Worker/Daemon.php (hunk 3 above)
Comment (annando) [reply src/Core/Worker/Daemon.php:53]: I guess so. This is just copied code.

**Review by MrPetovan — CHANGES_REQUESTED** [review]
Please fix the license issue.

**Comment by annando** [comment]
I guess that the license issue is a problem with the license check. The text is identical - but it doesn't start in the first line because it is a script.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**